x/auth/vesting/internal/types/codec.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting/exported"
+)
+
+// RegisterCodec registers concrete types on the codec
+func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterInterface((*exported.VestingAccount)(nil), nil)
+	cdc.RegisterConcrete(&BaseVestingAccount{}, "cosmos-sdk/BaseVestingAccount", nil)
+	cdc.RegisterConcrete(&ContinuousVestingAccount{}, "cosmos-sdk/ContinuousVestingAccount", nil)
+	cdc.RegisterConcrete(&DelayedVestingAccount{}, "cosmos-sdk/DelayedVestingAccount", nil)
+	cdc.RegisterConcrete(&PeriodicVestingAccount{}, "cosmos-sdk/PeriodicVestingAccount", nil)
+}
+
+// ModuleCdc module wide codec
+var ModuleCdc *codec.Codec
```

Comment:
vesting is not really a module, but a submodule
```suggestion
var SubModuleCdc *codec.Codec
```